var/spack/repos/builtin/packages/mpark-variant/package.py

```diff
@@ -5,19 +5,19 @@
 
 from spack import *
 
-
 class MparkVariant(CMakePackage):
     """C++17 `std::variant` for C++11/14/17"""
 
-    homepage = "https://mpark.github.io/variant"
-    url      = "https://github.com/mpark/variant/archive/v1.3.0.tar.gz"
+    homepage = "https://github.com/mpark/variant"
+    git     = "https://github.com/mpark/variant.git"
     maintainers = ['ax3l']
 
-    version('1.4.0', sha256='8f6b28ab3640b5d76d5b6664dda7257a4405ce59179220431b8fd196c79b2ecb')
-    version('1.3.0', sha256='d0f7e41f818fcc839797a8017e76b8b66b323651c304cff641a83a56ae9943c6')
+    version('1.4.0', tag='v1.4.0')
+    version('1.3.0', tag='v1.3.0')
 
     # Ref.: https://github.com/mpark/variant/pull/73
     patch('nvcc.patch', when='@:1.4.0')
+    patch('version.patch', when='@:1.4.0')
```

Comment:
Question for reviewer: we also have a patch that makes this work with Intel icpc and MacOSX (essentially on MacOSX with icpc the `#pragma __clang__` lies).  Are you open to applying that patch to 1.4.0 as well?

A more general question: they have bumped the version on master to 1.4.1, which has the CMake patch as well as some other fixes.  This is not officially tagged as a release on the mpark-variant Github.  Can we still add a `version(1.4.1)` with the git hash?  Or should we call its something else like `1.4.1-rc` for a release candidate?  Or just call it `master` and not make it the default? I am just not sure what the idiomatic spack way to do that is, but it would be nice to have those fixes available.

Reply:
Apple.patch file:
```
--- ./config.hpp 2021-01-26 09:42:45.000000000 -0700
+++ ./config.hpp.new 2021-01-26 09:42:23.000000000 -0700
@@ -50,7 +50,7 @@
 #define MPARK_BUILTIN_UNREACHABLE
 #endif

-#if __has_builtin(__type_pack_element)
+#if __has_builtin(__type_pack_element) &amp;&amp; !(defined(__ICC) &amp;&amp; defined(__APPLE__))
 #define MPARK_TYPE_PACK_ELEMENT
 #endif
```

Reply:
@ax3l , for some reason I couldn't figure out how to add you as a reviewer on this.

Reply:
Thanks! The patch looks good to me, do you like to send it upstream to https://github.com/mpark/variant so we can reference its PR link here?

Reply:
> they have bumped the version on master to 1.4.1, which has the CMake patch as well as some other fixes. This is not officially tagged as a release on the mpark-variant Github

Let's open an issue on mainline on this to inquire if they plan to tag it soon?

Reply:
I will PR it against mpark-variant.  We were hoping to hear back from Intel what the deal is with `__APPLE__` pragma issues before submitting the PR, but it seems unlikely to be resolved soon, and we are stuck with the intel 19 compilers we have now for probably a couple years.

Reply:
@ax3l I spend a PR on mpark-variant.  @azukaitis, the patch should not include the check on __APPLE__, only ICC I believe.  It occurred to me that if someone builds intel 19.0.4 with something like gcc 8, they could run into the same issue of the preprocessor claiming support for something Intel doesn't support.  I think this is acceptable since there is a narrow list of intel compilers that support all of this, it should have no performance effects, and it won't be long until we can just switch to using the Intel clang frontends.

Reply:
This PR does not contain the APPLE patch... that would be a separate issue I believe.

Reply:
It's ok for me to add this patch in the PR here as well, but I'll merge for now and you can do a follow-up with it :)

Reply:
I will push up the ICC patch on a new PR.